### PR TITLE
Use share code for get/set sync code and reset sync

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/11542

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/android/brave_sync_worker.cc [hunk 1]
```diff
@@ -210,28 +196,16 @@ bool BraveSyncWorker::IsFirstSetupComplete(
 bool BraveSyncWorker::ResetSync(
```
Comment (AlexeyBarabash) [inline browser/android/brave_sync_worker.cc:196]: this function should become `void` as it always returns  `true` now

**Review by AlexeyBarabash — APPROVED** [review]
LGTM

File: browser/android/brave_sync_worker.cc (hunk 1 above)
Comment (darkdh) [reply browser/android/brave_sync_worker.cc]: addressed in https://github.com/brave/brave-core/pull/6565/commits/73593a36b3f8013f983fe89691875c1ae00d88e0

**Review by AlexeyBarabash — APPROVED** [review]
LGTM

File: browser/android/brave_sync_worker.cc [hunk 2]
```diff
@@ -207,33 +189,19 @@ bool BraveSyncWorker::IsFirstSetupComplete(
          sync_service->GetUserSettings()->IsFirstSetupComplete();
 }
 
-bool BraveSyncWorker::ResetSync(
+void BraveSyncWorker::ResetSync(
     JNIEnv* env,
     const base::android::JavaParamRef<jobject>& jcaller) {
-  syncer::SyncService* sync_service = GetSyncService();
+  auto* sync_service =
```
Comment (bridiver) [inline browser/android/brave_sync_worker.cc:195]: nit - you could cast in `GetSyncService`

File: browser/ui/webui/settings/brave_sync_handler.cc [hunk 3]
```diff
@@ -83,12 +84,11 @@ void BraveSyncHandler::HandleGetSyncCode(const base::ListValue* args) {
...
   CHECK(args->Get(0, &callback_id));
 
-  brave_sync::Prefs brave_sync_prefs(profile_->GetPrefs());
-  std::string sync_code = brave_sync_prefs.GetSeed();
-  if (sync_code.empty()) {
-    std::vector<uint8_t> seed = brave_sync::crypto::GetSeed();
-    sync_code = brave_sync::crypto::PassphraseFromBytes32(seed);
-  }
+  auto* sync_service =
+    static_cast<syncer::BraveProfileSyncService*>(GetSyncService());
```
Comment (bridiver) [inline browser/ui/webui/settings/brave_sync_handler.cc:88]: lint nit: 4 spaces for continuation of a line

Comment (bridiver) [reply browser/ui/webui/settings/brave_sync_handler.cc:88]: also see https://github.com/brave/brave-core/pull/6565/files#r482274558

File: browser/android/brave_sync_worker.cc (hunk 2 above)
Comment (darkdh) [reply browser/android/brave_sync_worker.cc]: addressed in https://github.com/brave/brave-core/pull/6565/commits/3ce15e18d2629cc8017b6a39459249389f0d7cd8

File: browser/ui/webui/settings/brave_sync_handler.cc (hunk 3 above)
Comment (darkdh) [reply browser/ui/webui/settings/brave_sync_handler.cc]: addressed in https://github.com/brave/brave-core/pull/6565/commits/3ce15e18d2629cc8017b6a39459249389f0d7cd8

**Review by AlexeyBarabash — APPROVED** [review]
LGTM

**Comment by darkdh** [comment]
These known errors can be ignored:
```
15:00:30  + set -e
15:00:30  + /home/ubuntu/android-sdk/tools/mksdcard -l android29 10240M android.img
15:00:30  /home/ubuntu/workspace/pr-brave-browser-dedup-sync-handler-android@tmp/durable-e97aabad/script.sh: 3: /home/ubuntu/workspace/pr-brave-browser-dedup-sync-handler-android@tmp/durable-e97aabad/script.sh: /home/ubuntu/android-sdk/tools/mksdcard: not found
```
https://ci.brave.com/job/pr-brave-browser-dedup-sync-handler-android/10/execution/node/194/log/

```
15:21:55  powershell.exe : Start-Process : This command cannot be run due to the error: The system cannot find the file specified.
15:21:55  At C:\dler@tmp\durable-970b898f\powershellWrapper.ps1:3 char:1
15:21:55  + & powershell -NoProfile -NonInteractive -ExecutionPolicy Bypass -Comm ...
15:21:55  + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
15:21:55      + CategoryInfo          : NotSpecified: (Start-Process :...file specified.:String) [], RemoteException
15:21:55      + FullyQualifiedErrorId : NativeCommandError
15:21:55   
15:21:55  At C:\dler@tmp\durable-970b898f\powershellScript.ps1:6 char:17
15:21:55  + ...             Start-Process "C:\Users\Administrator\AppData\Local\Brave ...
15:21:55  +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
15:21:55  
15:21:55      + CategoryInfo          : InvalidOperation: (:) [Start-Process], InvalidOperationException
15:21:55  
15:21:55      + FullyQualifiedErrorId : InvalidOperationException,Microsoft.PowerShell.Commands.StartProcessCommand
```
https://ci.brave.com/job/pr-brave-browser-dedup-sync-handler-windows/8/execution/node/241/log/

**Pull request merged**